test/integration/multicluster/multicluster-traffic/mc_traffic_test.go

```diff
@@ -0,0 +1,369 @@
+	}
+
+	t.Run("install resources in target cluster", func(t *testing.T) {
+		// Create namespace in source cluster
+		out, err := TestHelper.KubectlWithContext("", contexts["src"], "create", "namespace", "linkerd-nginx-gateway-deploy")
+		if err != nil {
+			testutil.AnnotatedFatalf(t, "failed to create namespace", "failed to create namespace 'linkerd-nginx-gateway-deploy': %s\n%s", err, out)
+		}
+
+		out, err = TestHelper.KubectlApplyWithContext("", contexts["tgt"], "-n", "linkerd-nginx-gateway-deploy", "-f", "testdata/nginx-gateway-deploy.yaml")
+		if err != nil {
+			testutil.AnnotatedFatalf(t, "failed to install nginx deploy", "failed to install nginx deploy: %s\n%s", err, out)
+		}
+
+		// Wait for workloads to spin up in target cluster. These workloads will have
+		// their services mirrored in the source cluster. The test will check whether
+		// the gateway keeps track of the mirror services.
+		tgtWorkloadRollouts := map[string]testutil.DeploySpec{
+			"nginx-deploy": {Namespace: "linkerd-nginx-gateway-deploy", Replicas: 1},
+		}
+		TestHelper.WaitRolloutWithContext(t, tgtWorkloadRollouts, contexts["tgt"])
+	})
+
+	timeout := time.Minute
+	err := TestHelper.RetryFor(timeout, func() error {
+		out, err := TestHelper.LinkerdRun("--context="+contexts["src"], "multicluster", "gateways")
+		if err != nil {
+			return err
+		}
+		rows := strings.Split(out, "\n")
+		if len(rows) < 2 {
+			return errors.New("response is empty")
+		}
+		fields := strings.Fields(rows[1])
+		if len(fields) < 6 {
+			return fmt.Errorf("unexpected number of columns: %d", len(fields))
+		}
+		if fields[0] != "target" {
+			return fmt.Errorf("unexpected target cluster name: %s", fields[0])
+		}
+		if fields[1] != "True" {
+			return errors.New("target cluster is not alive")
+		}
+		if fields[2] != "1" {
+			return fmt.Errorf("invalid NUM_SVC: %s", fields[2])
+		}
+
+		return nil
+	})
+	if err != nil {
+		testutil.AnnotatedFatal(t, fmt.Sprintf("'linkerd multicluster gateways' command timed-out (%s)", timeout), err)
+	}
+}
+
+// TestCheckAfterRepairEndpoints calls `linkerd mc check` again after 1 minute,
+// so that the RepairEndpoints event has already been processed, making sure
+// that resyncing didn't break things.
+// CheckGateway will make use of nginx-gateway-deploy to see whether check
+// result returns number of mirror services.
+func TestCheckGatewayAfterRepairEndpoints(t *testing.T) {
+	// Re-build the clientset with the source context
+	if err := TestHelper.SwitchContext(contexts["src"]); err != nil {
+		testutil.AnnotatedFatalf(t, "failed to rebuild helper clientset with new context", "failed to rebuild helper clientset with new context [%s]: %v", contexts["src"], err)
+	}
+	time.Sleep(time.Minute + 5*time.Second)
+	cmd := []string{"--context=" + contexts["src"], "multicluster", "check", "--wait=10s"}
```

Comment:
We already switched the context to `src` right above this, so we shouldn't need to pass `--context` flag here.

Reply:
That's true, but the context switch above is meant for the Go clientset embedded in the TestHelper. When we apply resources or interact with Linkerd we're calling in to the binaries. Their context would be inferred from the environment and be independent of the clientset, so I think we still need to keep this in. Otherwise, linkerd/kubectl would use whatever config is in use in the environment at the time.

Reply:
Ah makes sense.